Terms/SolrRegexTermsService.php

```diff
@@ -24,10 +24,9 @@ public function fetchTerms(SimpleQueryInterface $query)
             return new EmptyTermsResult();
         }
 
-        $suggestQuery
-            ->setFields($field)
-            ->setRegex($query->getSearchTerm())
-            ->setSort('count');
+        $suggestQuery->setFields($field);
+        $suggestQuery->setRegex($this->escapeTerm($query->getSearchTerm()));
```

Comment:
Isn't this doing the same as the `escapeTerm` helper method?

This is where the method refers to regarding special characters though the link is dead
http://lucene.apache.org/core/queryparsersyntax.html#Escaping%20Special%20Characters

`preg_quote` doesn't work for the same reason.